framework/web/Request.php

```diff
@@ -169,26 +169,47 @@ class Request extends \yii\base\Request
      */
     private $_headers;
 
-
     /**
      * Resolves the current request into a route and the associated parameters.
      * @return array the first element is the route, and the second is the associated parameters.
+     * @redirect when an unresolved url matches an url with suffix 
      * @throws NotFoundHttpException if the request cannot be resolved.
      */
     public function resolve()
     {
-        $result = Yii::$app->getUrlManager()->parseRequest($this);
+        $currentQueryParams = $this->_queryParams;
+        if($url = $this->resolveRequest($this,$currentQueryParams)){
+            return $url;
+        }else if(isset(Yii::$app->getUrlManager()->suffix)){
+            $request = clone $this;
+            $request->setPathInfo($request->getPathInfo() . Yii::$app->getUrlManager()->suffix);                
+            if($url = $this->resolveRequest($request,$currentQueryParams)){
```

Comment:
This will produce significant performance impact, since URL rules will be parsed twice per each request processing.
Thus `preg_match()` calls will be doubled.